src/modules/navigator/rtl_direct.cpp

```diff
@@ -143,6 +150,23 @@ void RtlDirect::setRtlPosition(PositionYawSetpoint rtl_position, loiter_point_s
 			_destination.alt = _home_pos_sub.get().alt;
 		}
 
+		if (_global_pos_sub.get().terrain_alt_valid
```

Comment:
The destination is set to the current position only as long as it is not activated yet. What happens if we are below the hagl altitude on start, but during RTL we fly above because the terrain is now lower?
As soon as the hagl altitude is exceeded, it goes into failsafe descend and then into RTL again? Are we sure that the setRtlPosition is triggered at least once in between while it is above? If you make the check in the getActivationLandState function, we at least make sure that it is set correctly on activation. And you don't need the   _print_return_alt_higher_than_max_hagl variable